### PR TITLE
angular image should under assets and use MatIconModule

### PR DESCRIPTION
**Pull request by Tanya0609**
This change is for 1 of the angular potential improvements under experts' suggestion #681:
"The files under src/app/images should be moves to the assets folder src/assets and use the MatIconModule to register and use the SVG assets (see https://material.angular.io/components/icon/api)"

There is an example code of using MatIconModule under this link, which my code change is similar to.

**Summary to my change:**
1. move /images under /assets folder
2. utilize MatIconModule to render SVG assets
3. update package.json dependencies
4. since we are using Angular Material, I have to import one of the pre-build default themes to resolve the angular warning

**How to test:**
1. generate an angular project with at least Grid and Master Details pages selected 
2. start your project 

**Expected:**
The UI under Grid page and Master Details page should look the same as the before. Here attached screenshots about what the page should look like:
grid
master-detail

### CODE REVIEW

File: templates/Web/Pages/Angular.Grid/src/app/app-shell/AngularGrid/grid.component.ts [hunk 1]
```diff
@@ -25,7 +26,11 @@ export class GridComponent implements OnInit {
       id: 1
     }
   ];
-  constructor(private gridService: GridService) { }
+  constructor(private gridService: GridService, iconRegistry: MatIconRegistry, sanitizer: DomSanitizer) {
+    iconRegistry.addSvgIcon(
+      'grey-box',
+      sanitizer.bypassSecurityTrustResourceUrl('assets/GreyBox.svg'));
```
Comment (streetcoder123) [inline templates/Web/Pages/Angular.Grid/src/app/app-shell/AngularGrid/grid.component.ts:32]: What does `bypassSecurityTrustResourceUrl` mean?

Comment (Tanya0609) [reply templates/Web/Pages/Angular.Grid/src/app/app-shell/AngularGrid/grid.component.ts:32]: It is a default method. Cause this module uses HTTP Request to get svg images. Here it tries to make sure this image url won't have Cross Site Scripting Security issue when gets called. 

"DomSanitizer helps preventing Cross Site Scripting Security bugs (XSS) by sanitizing values to be safe to use in the different DOM contexts. Calling any of the bypassSecurityTrust... APIs disables Angular's built-in sanitization for the value passed in."
More details: https://angular.io/api/platform-browser/DomSanitizer

File: templates/Web/Pages/Angular.Grid/src/app/app-shell/AngularGrid/GridBox/grid-box.component.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.mat-icon {
```
Comment (piotrmark) [inline templates/Web/Pages/Angular.Grid/src/app/app-shell/AngularGrid/GridBox/grid-box.component.css:1]: Is this class used somewhere?

**Review by piotrmark — APPROVED** [review]
Looks good for me, but it would be nice if someone with Angular experience take a look

**Review by streetcoder123 — APPROVED** [review]
Code looks good to me though I do not understand it completely. But it works perfectly as it should 👍

File: templates/Web/Pages/Angular.Grid/src/app/app-shell/AngularGrid/GridBox/grid-box.component.css (hunk 2 above)
Comment (Tanya0609) [reply templates/Web/Pages/Angular.Grid/src/app/app-shell/AngularGrid/GridBox/grid-box.component.css:1]: This one did not use explicitly, but it is needed in order to use mat-icon.

**Review by annielovesjs — APPROVED** [review]
everything works as intended

**Pull request merged**